writers/batchwriter/batchwriter.go

```diff
@@ -325,9 +325,9 @@ func (w *BatchWriter) startWorker(ctx context.Context, msg *message.WriteInsert)
 	return nil
 }
 
-func timer(timeout time.Duration) <-chan time.Time {
-	if timeout == 0 {
-		return nil
+func ticker(interval time.Duration) (ch <-chan time.Time, stop func()) {
+	if t := time.NewTicker(interval); t != nil {
```

Comment:
0 interval would panic:
```go
func NewTicker(d Duration) *Ticker {
	if d <= 0 {
		panic(errors.New("non-positive interval for NewTicker"))
	}
```

Reply:
So something like `if interval == 0 { // create and empty channel and return it, stop fn to close the empty channel` is needed

Reply:
68015d39fd8c4ab30b82969e283a2651cedcfe8b